<launch> 

   <!-- parameters-->
   <arg name="destination_port" default="11021" doc="ADMA Webinterface -> Data -> Ethernet data output -> Destination Port" />
   <arg name="destination_ip" default="192.168.88.255" doc="ADMA Webinterface -> Data -> Ethernet data output -> Destination IP" />
   <arg name="use_performance_check" default="False" doc="True if you want to get additional performance logging" />
   <arg name="protocol_version" default="v3.3.4" doc="The desired protocol version of the ADMA UDP data packets" />
   <arg name="mode" default="default" doc="the desired execution mode, options are default/record/replay" />
   <arg name="frame_id_navsatfix" default="gnss_link" doc="ROS Frame-ID of NavSatFix msg" />
   <arg name="frame_id_imu" default="imu_link" doc="ROS Frame-ID of Imu msg" />
   <arg name="frame_id_adma" default="adma" doc="ROS Frame-ID of ADMA data msg" />
   <arg name="frame_id_adma_status" default="adma_status" doc="ROS Frame-ID of ADMA Status msg" />
   <arg name="frame_id_data_raw" default="data_raw" doc="ROS Frame-ID of ADMA raw data msg" />
   
   <!-- start ADMA Driver node -->
   <node name="adma_driver" type="driver_new" pkg="adma_ros_driver" output="screen">
      <!-- add parameters -->
      <param name="/destination_port" type="int" value="$(arg destination_port)"/>
      <param name="/destination_ip" type="string" value="$(arg destination_ip)"/>
      <param name="/use_performance_check" type="bool" value="$(arg use_performance_check)"/>
      <param name="/protocol_version" type="string" value="$(arg protocol_version)"/>
      <param name="/mode" type="string" value="$(arg mode)"/>
      <param name="/frame_id_navsatfix" type="string" value="$(arg frame_id_navsatfix)"/>
      <param name="/frame_id_imu" type="string" value="$(arg frame_id_imu)"/>
      <param name="/frame_id_adma" type="string" value="$(arg frame_id_adma)"/>
      <param name="/frame_id_adma_status" type="string" value="$(arg frame_id_adma_status)"/>
      <param name="/frame_id_data_raw" type="string" value="$(arg frame_id_data_raw)"/>
      <!-- allow remapping of topics (modify the to="" to publish with other topics)-->
      <remap from="adma/data" to="adma/data"/>
      <remap from="adma/data_scaled" to="adma/data_scaled"/>
      <remap from="adma/data_raw" to="adma/data_raw"/>
      <remap from="adma/status" to="adma/status"/>
      <remap from="adma/heading" to="adma/heading"/>
      <remap from="adma/velocity" to="adma/velocity"/>
      <remap from="adma/fix" to="adma/fix"/>
      <remap from="adma/imu" to="adma/imu"/>
   </node>

   <!-- arg to activate the gsdb logging (ADMA-PP)-->
   <arg name="log_gsdb" default="true" doc="True if you want to create a gsdb logfile for post-processing" />
   <node name="bag2gsdb" type="bag2gsdb_converter" pkg="adma_tools_cpp" output="screen" if="$(arg log_gsdb)">
      <remap from="adma/data_recorded" to="adma/data_raw"/>
   </node>
   <!-- arg to activate rosbag recording -->
   <arg name="record_rosbag" default="false" doc="True if you want to create a rosbag" />
   <node name="rosbag_record_raw_data" type="record" pkg="rosbag" args="record -a" if="$(arg record_rosbag)" />
   
</launch>
